File: backend/src/test/java/com/skewwhiffy/auraltester/notation/factory/NoteFactoryTest.java
package com.skewwhiffy.auraltester.notation.factory;

import com.skewwhiffy.auraltester.notation.model.key.Key;
import com.skewwhiffy.auraltester.notation.model.note.AbsoluteNote;
import lombok.val;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.mockito.InjectMocks;
import org.mockito.junit.jupiter.MockitoExtension;

import static org.assertj.core.api.Assertions.assertThat;

@ExtendWith(MockitoExtension.class)
public class NoteFactoryTest {
    @InjectMocks
    private NoteFactory noteFactory;

    @Test
    void instantiatesMiddleC() {
        val expected = AbsoluteNote.getMiddleC();

        val actual = noteFactory.getAbsoluteNote("C");

        assertThat(actual).isEqualTo(expected);
    }

    @Test
    void instantiatesNoteAboveMiddleC() {
        val expected = "c''";

        val actual = noteFactory.getAbsoluteNote(expected);

        assertThat(actual.getAbc(Key.getCMajor())).isEqualTo(expected);
    }
    /*

  @Test
  fun `instantiate note below middle C`() {
    val internalNotation = "Dx#,,,"
    val expected = "^^^D,,,"

    val actual = noteFactory.getAbsoluteNote(internalNotation)

    assertThat(actual.abc(Key.cMajor)).isEqualTo(expected)
  }

  @Test
  fun `throw when note name invalid`() {
    assertThrows<IllegalArgumentException> {
      noteFactory.getAbsoluteNote("H")
    }
  }
}
     */
}

File: backend/src/test/kotlin/com/skewwhiffy/auraltester/notation/factory/NoteFactoryTest.kt
package com.skewwhiffy.auraltester.notation.factory

/*
import com.skewwhiffy.auraltester.notation.model.key.Key
import lombok.`val`
import org.assertj.core.api.Assertions
import org.junit.jupiter.api.Test

@ExtendWith(MockitoExtension::class)
class NoteFactoryTest {
    @InjectMocks
    private val noteFactory: NoteFactory? = null
    @Test
    fun instantiatesMiddleC() {
        val expected: `val` = AbsoluteNote.getMiddleC()
        val actual: `val` = noteFactory!!.getAbsoluteNote("C")
        Assertions.assertThat<`val`>(actual).isEqualTo(expected)
    }

    @Test
    fun instantiatesNoteAboveMiddleC() {
        val expected: `val` = "c''"
        val actual: `val` = noteFactory!!.getAbsoluteNote(expected)
        assertThat(actual.getAbc(Key.getCMajor())).isEqualTo(expected)
    }

    @Test
    fun instantiatesNoteBelowMiddleC() {
        val internalNotation: `val` = "Dx#,,,"
        val expected: `val` = "^^^D,,,"
        val actual: `val` = noteFactory!!.getAbsoluteNote(internalNotation)
        assertThat(actual.getAbc(Key.getCMajor())).isEqualTo(expected)
    }

    @Test
    fun when_noteNameInvalid_then_throws() {
        Assertions.assertThatThrownBy(ThrowingCallable { noteFactory!!.getAbsoluteNote("H") })
            .isInstanceOf(IllegalArgumentException::class.java)
    }
}


 */